src/hotspot/share/nmt/virtualMemoryTracker.hpp

```diff
@@ -194,15 +194,38 @@ class VirtualMemorySummary : AllStatic {
  */
 class VirtualMemoryRegion {
  private:
-  address      _base_address;
-  size_t       _size;
+  address         _base_address;
+  size_t          _size;
+  MemTag          _mem_tag;
+  NativeCallStack _reserved_stack;
+  NativeCallStack _committed_stack;
 
  public:
+  VirtualMemoryRegion() :
+    _base_address(0), _size(0), _mem_tag(mtNone),
+    _reserved_stack(NativeCallStack::empty_stack()) ,
+    _committed_stack(NativeCallStack::empty_stack()) {}
+
   VirtualMemoryRegion(address addr, size_t size) :
-    _base_address(addr), _size(size) {
+    _base_address(addr), _size(size), _mem_tag(mtNone),
+    _reserved_stack(NativeCallStack::empty_stack()) ,
+    _committed_stack(NativeCallStack::empty_stack()) {
      assert(addr != nullptr, "Invalid address");
      assert(size > 0, "Invalid size");
-   }
+  }
+
+  VirtualMemoryRegion(address addr, size_t size, const NativeCallStack& stack, bool committed) :
```

Comment:
Correct me if I'm wrong but is this `committed` only ever true? And then we use the presence of this constructor parameter to decide whether we are creating a committed or a reserved region?

If so, at the very least we should assert that committed is true. But I think it would be nicer to make the constructor private and make some static factory methods akin to `VirtualMemoryRegion::committed` and `VirtualMemoryRegion::reserved`.